### PR TITLE
Update explanation for metrics of AllenNLPExecutor

### PR DESCRIPTION
**Pull request by himkt**
This PR tries to improve the documentation for AllenNLP integration.

### CODE REVIEW

**Comment by himkt** [comment]
@chenghuzi I heard that you find the docs of `metrics` parameter in AllenNLP integration vague and difficult to understand (sorry). I tried to make the explanation more informative... Could you please give me a feedback?

**Comment by hvy** [comment]
@HideakiImamura could you help reviewing this PR?

File: optuna/integration/allennlp/_executor.py [hunk 1]
```diff
@@ -101,7 +101,11 @@ class AllenNLPExecutor(object):
         serialization_dir:
             A path which model weights and logs are saved.
         metrics:
-            An evaluation metric for the result of ``objective``.
+            An evaluation metric. `Trainer <https://docs.allennlp.org/
+            main/api/training/gradient_descent_trainer/#train>`_ of AllenNLP
+            returns a dictionary consists of metrics after training.
```
Comment (hvy) [inline optuna/integration/allennlp/_executor.py:106]: A grammatical suggestion.
```suggestion
            returns a dictionary containing metrics after training.
```

**Review by hvy — CHANGES_REQUESTED** [review]
Thanks, looks like an improvement except I left a small nit.

On a side note, I found it a bit confusing that the text is "Trainer" while the actual link targets "GradientDescentTrainer.train", but this is clear from context to expected users?

**Review by hvy — APPROVED** [review]
LGTM!

**Comment by himkt** [comment]
> I found it a bit confusing that the text is "Trainer" while the actual link targets "GradientDescentTrainer.train", but this is clear from context to expected users?

Thank you for the comment. In AllenNLP, `Trainer` is a base class and `GradientDescentTrainer` is used in most cases. But users with AllenNLP CLI may not be aware of `GradientDescentTrainer` because it is hidden inside `allennlp train` command. At least users may understand AllenNLP train a model using some trainer so I used the term `Trainer`. 💭 But based on your comment, I agree that users may be confusing for that. I'll revise the PR to show `GradientDescentTrainer` in the text. 😸

**Review by hvy — APPROVED** [review]
Thanks, LGTM!

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Comment by chenghuzi** [comment]
> @chenghuzi I heard that you find the docs of `metrics` parameter in AllenNLP integration vague and difficult to understand (sorry). I tried to make the explanation more informative... Could you please give me a feedback?

Sorry for the delayed response. The new explanation looks much more clear to me. Thanks!

**Pull request merged**